pkg/controller/build/build_controller.go

```diff
@@ -39,8 +39,77 @@ const succeedStatus string = "Succeeded"
 const namespace string = "namespace"
 const name string = "name"
 
-type setOwnerReferenceFunc func(owner, object metav1.Object, scheme *runtime.Scheme) error
+// error message
+const (
+	// validate secrets
+	InitialErrorCodes = 0000*1 + iota
```

Comment:
@zhangtbj im wondering if we can better replace the <"B" + CODE> approach in favor of just `ints`. For fulfilling the distinction of Build vs any other CRD that requires error codes in the future. Therefore I will propose to use the following:
`1000*1 + iota` => For Builds
`2000*1 + iota` => For BuildRuns

The above simplify the code logic on prepending a "B" string.

Reply:
This is how it looks now, seems there is a bug:

```sh
$ k -n test-build get build
NAME                  REGISTERED   REASON                             BUILDSTRATEGYKIND      BUILDSTRATEGYNAME   CREATIONTIME
kaniko-golang-build   False        B: secret foobar does not exist   ClusterBuildStrategy   kaniko              2s
```

I would propose this to look like:

```
$ k -n test-build get build
NAME                  REGISTERED   REASON                              BUILDSTRATEGYKIND      BUILDSTRATEGYNAME   CREATIONTIME
kaniko-golang-build   False        1003: secrets foobar do not exist   ClusterBuildStrategy   kaniko              2s
```

Reply:
I am ok for this approach. It is easy for parsing.